packages/nextjs/mocks/msw/msw-enable.ts

```diff
@@ -0,0 +1,42 @@
+import { handlers } from "./graphql";
+import { groqHandlers } from "./groq/handlers";
+
+/**
+ * Importing this file enables MSW.  It works both client-side and server-side.
+ */
+
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+// See https://github.com/mswjs/interceptors/issues/159
+const mswUnsupported = process.version.startsWith("v18");
+if (mswUnsupported) {
+  throw new Error(`[MSW Mock Server] MSW does not support mocking 'fetch' in Node v18`);
+}
+
+async function initializeMSW() {
+  console.log("[MSW Mock Server] Starting...");
+  if (typeof window === "undefined") {
+    // 'next dev' will compile pages individually, so let's ensure we only install MSW once:
+    const isPatchedAlready = Object.getOwnPropertySymbols(require("http")).find(
+      (s) => s.toString() === "Symbol(isPatchedModule)"
+    );
```

Comment:
🤔 This is interesting.  How would this present itself?  Will it be an error, or just an unnecessary memory overhead the more you use the site?  Should I look into doing this to my project as well?

Reply:
Yes, this was some bizarre behaviour that I was seeing. I didn't know that `next dev` would behave this way. It seems like all pages are run in child processes, or something like that, so they don't share the "require" cache.
I'm not totally sure if "multiple msw" caused problems, or if they simply bypassed each other. I added this when I was debugging some performance issues, but it didn't solve it, so I don't know if it really matters. Things seemed to work either way.